src/crewai/agents/executor.py

```diff
@@ -106,15 +106,16 @@ def _iter_next_step(
                 **inputs,
             )
             if self._should_force_answer():
```

Comment:
@joaomdmoura please review specifically this.

Reply:
I thinks it's missing the `isinstance(output, AgentAction) or isinstance(output, AgentFinish):`

Reply:
does this covers `isinstance(output, AgentFinish)`?

Reply:
Don't think it's needed. It will fallback to `yield output` and `return`. We only need to force answer if the type is either `AgentAction` or a list of `AgentAction`. Does that make sense?